### PR TITLE
Allow peeking z levels.

### PR DESCRIPTION
**Pull request by OzoneH3**
#### Summary
```SUMMARY: Features "Allow peeking z levels."```

Allows you to peek Up and Down zlevels.

Uses the functionality of ```vertical_move()``` to determine if the player can move up/down and where he will be positionwise if it's stairs.

Otherwise it's using the same principle as ```peek()```. Activates ```look_around()``` after a successful move and moves back to the previous position after the exit of ```look_around()```.

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -6400,12 +6400,23 @@ void game::examine( const tripoint &examp )
...
 void game::peek()
 {
-    const cata::optional<tripoint> p = choose_adjacent( _( "Peek where?" ) );
+    const cata::optional<tripoint> p = choose_adjacent( _( "Peek where?" ), true );
     if( !p ) {
         refresh_all();
         return;
     }
 
+    if( ( *p ).z != 0 ) {
```
Comment (BevapDin) [inline src/game.cpp]:
```suggestion
    if( p->z != 0 ) {
```
Same for all the code below.

**Comment by mlangsdorf** [comment]
Related feature request: scroll through map memory on different levels.

**Comment by nexusmrsep** [comment]
**If** this means never again walking into a turret room above or below then the rest is self explanatory I guess.

**Comment by ralreegorganon** [comment]
I'm seeing some incorrect behavior in the placement location between peeking and actually moving.

World settings are all default, except experimental Z-levels are enabled.

**start location**
screen shot 2018-12-11 at 09 17 10

**moving up**
screen shot 2018-12-11 at 09 17 22

**peeking up**
screen shot 2018-12-11 at 09 17 31

Also, this does spend moves after the look_around is executed (vertical_move spends moves and is called before and after the look around), so it's different than the horizontal peek (where all moves are expended before peeking) which brings up some of the things I mentioned in my blind throwing PR [here](https://github.com/CleverRaven/Cataclysm-DDA/pull/26949#issuecomment-444981679): 

> This would have the side effect of making it so that creatures would move after the player finished taking a look/throwing, rather than before.

> I don't know that it'd be wrong, but I think it would be unexpected given current behavior where I know that when I exit the peek, everything is in the state I just saw. If my peek revealed a monster right around the corner, I know I can start running immediately and the distance between us will be what I just saw. If the move point deduction comes after the peek, then I may exit the peek only to be attacked because the monster advanced around the corner and attacked me while I was exiting the peek.

I think to this comment: 

> If this means never again walking into a turret room above or below then the rest is self explanatory I guess.

I think as long as you're not slowed down too much so the move deduction after peeking doesn't make you sit in front of the turret, you're ok, but unlike the horizontal peek there's still the chance of getting popped.

**Comment by OzoneH3** [comment]
Turns out that stair alignment is still a huge pita. Closed for now.

**Comment by OzoneH3** [comment]
Nevermind, was just a bug at z -1... vertical move is working as expected.

@ralreegorganon
> Also, this does spend moves after the look_around is executed (vertical_move spends moves and is called before and after the look around), so it's different than the horizontal peek (where all moves are expended before peeking)

Peeking is like stopping time and after it's finished, time catches up to all the things you did while peeking. Nothing is processed in the middle of all that.

That's the same reason why having a grenade with 3 turns going off after only one turn after blind throwing is a thing as i mentioned there.

**Comment by ralreegorganon** [comment]
This works in my test case now 👍

**Pull request merged**